### DISCUSSION TITLE
[Installation on Ubuntu]

### DISCUSSION THREAD

**Discussion by Advaith2212** [NVlabs/sionna#296] [Q&A] [answered]
Hey all, 

I have some difficulty in installing `sionna` in a new desktop.

The ubuntu version is as follows:
`Distributor ID:	Ubuntu`
`Description:	Ubuntu 22.04.3 LTS`
`Release:	22.04`

I created a new environment in conda with `python=3.8` and followed the command 
`pip install sionna`, which installed sionna version `0.16.1` and tensorflow `2.13.1`

When I tried to verify the installation, it says Skipping registering GPU devices and throws a mitsuba error. 

`AttributeError: jit_init_thread_state(): the LLVM backend is inactive because the LLVM shared library ("libLLVM.so") could not be found! Set the DRJIT_LIBLLVM_PATH environment variable to specify its path`

Can you help me with this

Thanks in advance

**Comment by vahidirp** [comment]
It looks like you're encountering an issue related to the LLVM library while trying to use Sionna. The error message suggests that the LLVM shared library ("libLLVM.so") could not be found. To resolve this issue, you can try the following steps:



Install LLVM:
Ensure that LLVM is installed on your system. You can install it using the package manager. For Ubuntu, you can use the following command:
```
sudo apt-get install libllvm-<version>-dev
```
Replace <version> with the appropriate version number. You can search for the available versions using:

```
apt search libllvm
```
After installing LLVM, try running your code again.

Set the DRJIT_LIBLLVM_PATH Environment Variable:
If installing LLVM doesn't resolve the issue, you may need to set the DRJIT_LIBLLVM_PATH environment variable. You can set it by exporting the variable in your shell. Open your terminal and run:

```
export DRJIT_LIBLLVM_PATH=/path/to/your/libLLVM.so
```
Replace /path/to/your/libLLVM.so with the actual path to the libLLVM.so file on your system.

Check Mitsuba Version Compatibility:
Ensure that the version of Mitsuba you are using is compatible with the version of Sionna you installed. Sometimes, there may be compatibility issues between different versions of libraries.

Update Sionna and TensorFlow:
It's a good idea to make sure you are using the latest versions of Sionna and TensorFlow. You can update them using:
```
    pip install --upgrade sionna tensorflow
```
    This will install the latest available versions.

After trying these steps, attempt to run your code again and see if the issue persists. If the problem persists, you may want to check the Sionna and Mitsuba documentation or community forums for more specific information related to the error you're encountering.

**Reply by merlinND** [reply, marked as answer]
Thanks for the summary of possible solutions @vahidirp.
@Advaith2212, feel free to also check out existing discussions mentioning this error message: https://github.com/NVlabs/sionna/discussions?discussions_q=DRJIT_LIBLLVM_PATH

@Advaith2212 To confirm, does your machine have a CUDA-compatible GPU available or not? LLVM should only be required when _not_ using the GPU.

**Reply by jiadong5** [reply]
Hi, I wonder how can I determine the appropriate version of llvm?

**Reply by merlinND** [reply]
Hello @jiadong5, in principle you should be able to install any recent version of LLVM. For example, LLVM 19 works fine on my machine.

**Reply by jiadong5** [reply]
Thx
